### ISSUE TITLE
Issues using Img2Img + ControlNet

### ISSUE THREAD

**Issue by potesd** [olegchomp/TDDiffusionAPI#1]
Hey there! I've not yet been able to get Img2Img work using your TDDiffusion API, I've got it working in a few other local API systems, but I'm desperately trying to access ControlNet and it's not working. As of right now if I pulse dependencies, it says "ControlNet Disabled" and when I try to connect a TOP to the Img2Img input, it fails to do anything. It gets stuck on Processed 0/11 steps and doesn't save any images or ever finish. 

I'm not getting any errors, just a failure to process. I'm still able to process in the WebUI no problem. 

Edit:
I've since gotten Img2Img working, but ControlNet is still saying "Disabled".

**Comment by olegchomp** [comment]
Hey! Did you install controlnet in a1111? Because TDA is only interface that works throw API with it. 
"Disabled" means that component can't find API endpoint for controlnet. 
btw. for now controlnet works only with txt2img, you can pass any image as a source for it in "controlnet input" TOP.

**Comment by olegchomp** [comment]
Controlnet img2img added in v.2.0.3

**Comment by zeroxoxo** [comment]
> Hey there! I've not yet been able to get Img2Img work using your TDDiffusion API, I've got it working in a few other local API systems, but I'm desperately trying to access ControlNet and it's not working. As of right now if I pulse dependencies, it says "ControlNet Disabled" and when I try to connect a TOP to the Img2Img input, it fails to do anything. It gets stuck on Processed 0/11 steps and doesn't save any images or ever finish.
> 
> I'm not getting any errors, just a failure to process. I'm still able to process in the WebUI no problem.
> 
> Edit: I've since gotten Img2Img working, but ControlNet is still saying "Disabled".

how did you fix img2img? i have the same problem
edit: nevermind switching to latest touchdesigner version fixed it

**Comment by soundprayer** [comment]
Tried on two a1111 installations, everything up to date. Tried to change some python code in TDDiffusion, but with no effect. txt2img works with controlnet, but img2img not.

[ControlNet] warning: using deprecated '/controlnet/img2img' route
[ControlNet] warning: consider using the '/sdapi/v1/img2img' route with the 'alwayson_scripts' json property instead
Error running process: C:\Users\user\Documents\A1111 Web UI Autoinstaller\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\controlnet.py
Traceback (most recent call last):
  File "C:\Users\user\Documents\A1111 Web UI Autoinstaller\stable-diffusion-webui\modules\scripts.py", line 417, in process
    script.process(p, *script_args)
  File "C:\Users\user\Documents\A1111 Web UI Autoinstaller\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\controlnet.py", line 701, in process
    unit = self.parse_remote_call(p, unit, idx)
  File "C:\Users\user\Documents\A1111 Web UI Autoinstaller\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\controlnet.py", line 612, in parse_remote_call
    unit.enabled = selector(p, "control_net_enabled", unit.enabled, idx, strict=True)
AttributeError: 'str' object has no attribute 'enabled'

I found two links talking abut solution: https://github.com/Mikubill/sd-webui-controlnet/issues/567
but I am not experienced enough with json. Will try again soon

**Comment by olegchomp** [comment]
@soundprayer what is your TDA version?

**Comment by olegchomp** [comment]
@soundprayer will fix that in next update.
for now you can comment line 109 in TdaEXT DAT
`#endpoint = '/controlnet/img2img'`

**Comment by soundprayer** [comment]
...to temporarily disable controlnet. Ok, thank you :)

**Comment by olegchomp** [comment]
its different controlnet endpoint. If you comment line, the right one should be appended to request.

**Comment by soundprayer** [comment]
hmm, after change it just continues to work, but without controlnet (I am checking the A1111 console). Before commenting it was giving me errors.

**Comment by chrissnea** [comment]
Hello there, I followed your tutorial to install everything, but unfortunately I am getting bunch of errors and in the TdaEXT and I know nothing about programming to know what's going on. I am not able to run any option. When I open the tox file the base error says: 
Script errors:
Error: (Extension 1)   File"/TDDiffisionAPI/TdaEXT", line 20, in __init__
File "/TDDiffusionAPI/TdaEXT" ,  line 383, in prepare 
td.AttributeError: 'tdbaseComp' object has no attribute  'WebClient' Context: /sys/TDResources (/TDDiffusionAPI)

I dont know if it is relevant but unlike your webui_user bat file mine wrote: set COMMANDLINE_ARGS= --no-half 
so I kept it like that and added the --api afterwards.
Thanks in advance.
